backend/node/base.py

```diff
@@ -16,7 +17,7 @@ def __init__(
         provider: str,
         model: str,
         config: dict,
-        leader_receipt: Optional[dict] = None,
+        leader_receipt: Optional[Receipt] = None,
     ):
         self.validator_mode = validator_mode
```

Comment:
on some parts of the code `validator_mode` is a string, and in others it's an `ExecutionMode` enum. I think they should all be `ExecutionMode`. 

Be careful on places where the serialization to `str` is required (like when returning a JSON). Remember to cast the enum to its `.value` on these "endpoints"

Reply:
Done!